src/components/select/select.js

```diff
@@ -316,20 +315,23 @@ function SelectDirective($mdSelect, $mdUtil, $mdTheming, $mdAria, $rootElement,
       element.attr(ariaAttrs);
 
       scope.$on('$destroy', function() {
-        if (isOpen) {
-          $mdSelect.hide().finally(function() {
```

Comment:
the `$mdSelect.destroy()` calls the $$interimElement.destroy which in turn calls 

```js
/*
       * Special method to quick-remove the interim element without animations
       */
      function destroy() {
        var interim = stack.shift();

        return interim ? interim.remove(SHOW_CANCELLED, false, {'$destroy':true}) :
               $q.when(SHOW_CANCELLED);
      }
```

which eventually calls `onRemove()`. None of this has changed.